### PR TITLE
Casper provisioning & deployment for OpenLMIS

### PR DESCRIPTION
**Pull request by pgarrison**
* Adds .swp and .swo to `.gitignore` for vim
* Adds terraform and deployment environments for casper
* Adds variables `bill-to`, `app-instance-ssh-key-name`, `app-use-route53-domain`, `app-route53-zone-name` and `aws-tls-cert-domain`. Defaults are set to old hard-coded values.
* The ARN for the TLS certificate was hardcoded. Now, it is automatically determined by querying AWS with the domain name the cert is for, `aws-tls-cert-domain` (with wildcards or other DNS syntax).

### CODE REVIEW

**Comment by pgarrison** [comment]
For benin-demo, demo-v3, and functional-test, `terraform plan` shows only minimal unrelated changes. UAT2 however would have more changes, but it seems most of that is not caused by this PR (though it would add BillTo: OpenLMIS).

**Comment by chunky56** [comment]
@pgarrison @joshzamor we should decide if we're using an RDS instance or the Docker database container for the database, and modify the provisioning/deployment config accordingly. I'm leaning towards the db container, especially since it's already at 9.6, and will make comments to that effect.

File: provision/terraform/uat/casper/terraform.tfvars [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+docker-tls-port = "2376"
+
+app-dns-name = "casper.a.openlmis.org"
+app-use-route53-domain = true
+app-route53-zone-name = "a.openlmis.org"
+aws-tls-cert-domain = "*.a.openlmis.org"
+
+app-instance-group = "docker-hosts"
+
+olmis-db-instance-class = "db.t2.medium"
```
Comment (chunky56) [inline provision/terraform/uat/casper/terraform.tfvars:19]: This and the two entries below it should probably be cleared (similar to demo-v3) if we are using the db container.

**Review by chunky56 — DISMISSED** [review]
Assuming we are using the db container for casper like demo-v3 uses, we should make these changes.

**Comment by chunky56** [comment]
On second thought, let's just go with the RDS configuration here, since you got it working, and it's our recommended deployment topology. http://docs.openlmis.org/en/latest/deployment/topology.html#recommended-deployment-topology

**Comment by chunky56** [comment]
What would be ideal, is to remove the db Docker image information then. I'll leave that to you to do in a simple commit to master.

**Comment by pgarrison** [comment]
Removed db Docker image in [c4df64af](https://github.com/OpenLMIS/openlmis-deployment/commit/c4df64af308fec93f20df5b0e8c7420239d1e7eb)

**Pull request merged**